### PR TITLE
Hotfix/int 764 bug unable to create pr

### PR DESCRIPTION
**Pull request by DmytroHryshyn**

### CODE REVIEW

File: intuita-webview/src/sourceControl/App.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ const getViewComponent = (view: View) => {
 		case 'createIssue': {
 			return <CreateIssue {...view.viewProps} />;
 		}
-		case 'createPR':
+		case 'createOrUpdatePR':
```
Comment (grzpab) [inline intuita-webview/src/sourceControl/App.tsx:31]: how about `upsertPR`? or `upsertPullRequest` (I prefer not to use acronyms if we could replace them with short names).

File: intuita-webview/src/sourceControl/CreatePRView/index.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ type Props = {
 	initialFormData: Partial<FormData>;
 	baseBranchOptions: string[];
 	targetBranchOptions: string[];
+	PRAlreadyExists: boolean;
```
Comment (grzpab) [inline intuita-webview/src/sourceControl/CreatePRView/index.tsx:17]: is `pullRequestAlreadyExists` better?

File: intuita-webview/src/sourceControl/CreatePRView/index.tsx [hunk 3]
```diff
@@ -100,32 +102,40 @@ const CreatePR = ({
...
-					}
-				>
-					Body
-				</VSCodeTextArea>
+				{!PRAlreadyExists ? (
+					<>
+						<VSCodeTextField
+							placeholder="title"
+							value={title}
+							onInput={(e) =>
```
Comment (grzpab) [inline intuita-webview/src/sourceControl/CreatePRView/index.tsx:110]: can you put this function as a named function in the body? The level of indentation here is pretty high.

File: intuita-webview/src/sourceControl/CreatePRView/index.tsx [hunk 4]
```diff
@@ -100,32 +102,40 @@ const CreatePR = ({
...
+									title: (e.target as HTMLInputElement).value,
+								})
+							}
+						>
+							Title
+						</VSCodeTextField>
+						<VSCodeTextArea
+							placeholder="Description"
+							value={body}
+							onInput={(e) =>
```
Comment (grzpab) [inline intuita-webview/src/sourceControl/CreatePRView/index.tsx:122]: same with the function here perhaps?

File: src/components/webview/repository.ts [hunk 5]
```diff
@@ -74,12 +74,35 @@ export class RepositoryService {
...
+
+		try {
+			return this.__repo.getBranch(branchName);
+		} catch (e) {
+			return null;
+		}
+	}
+
+	public async isBranchExists(branchName: string): Promise<boolean> {
+		return (await this.getBranch(branchName)) !== null;
```
Comment (grzpab) [inline src/components/webview/repository.ts:90]: I'd split this line into two, it's hard to read what's going on here.

File: src/extension.ts [hunk 6]
```diff
@@ -307,6 +309,7 @@ export async function activate(context: vscode.ExtensionContext) {
 						},
 						loading: false,
 						error: '',
+						PRAlreadyExists: Boolean(PR),
```
Comment (grzpab) [inline src/extension.ts:312]: what's the return type of `PR`?

File: src/components/webview/sourceControl.ts [hunk 7]
```diff
@@ -16,6 +16,13 @@ type CreatePRResponse = {
 	html_url: string;
 };
 
+type ListPRResponse = {
+	html_url: string;
+	head: {
```
Comment (grzpab) [inline src/components/webview/sourceControl.ts:21]: I think we don't need `head`?

File: src/components/webview/sourceControl.ts [hunk 8]
```diff
@@ -90,4 +97,29 @@ export class SourceControlService {
...
+		}
+
+		const result = await axios.get<ListPRResponse>(
+			'https://telemetry.intuita.io/sourceControl/github/pulls',
+		);
+
+		return result.data;
+	}
+
+	async getPRForBranch(branchName: string) {
```
Comment (grzpab) [inline src/components/webview/sourceControl.ts:121]: can you type the return type here? I think the type should is `Promise<ListPRResponse | null>`

File: src/components/webview/sourceControl.ts [hunk 9]
```diff
@@ -90,4 +97,29 @@ export class SourceControlService {
...
+		const PRList = await this.listPR();
+		return PRList.find((pr) => pr.head.ref === branchName);
```
Comment (grzpab) [inline src/components/webview/sourceControl.ts:123]: I'd consider using `?? null` at the end (`undefined` makes little sense for a list that cannot contain null values)

File: src/extension.ts [hunk 10]
```diff
@@ -358,12 +361,14 @@ export async function activate(context: vscode.ExtensionContext) {
 							decoded.right.targetBranch,
 						);
 
-						const { html_url } = await sourceControl.createPR(
-							decoded.right,
-						);
+						const { html_url } =
```
Comment (grzpab) [inline src/extension.ts:364]: I would split this into couple of statements

File: src/extension.ts (hunk 6 above)
Comment (DmytroHryshyn) [reply src/extension.ts:312]: its PullRequest | undefined

File: src/components/webview/sourceControl.ts (hunk 7 above)
Comment (DmytroHryshyn) [reply src/components/webview/sourceControl.ts:21]: we need it to find a PR for specific branch. `head.ref` is the branch name

File: intuita-webview/src/sourceControl/CreatePRView/index.tsx [hunk 11]
```diff
@@ -56,6 +58,17 @@ const CreatePR = ({
 		}));
 	}, [initialFormData]);
 
+	const onChangeFormField =
+		(fieldName: string) => (e: Event | React.FormEvent<HTMLElement>) => {
```
Comment (grzpab) [inline intuita-webview/src/sourceControl/CreatePRView/index.tsx:62]: what's the problem with the types here?

**Review by grzpab — APPROVED** [review]

**Pull request merged**